src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -48,238 +51,314 @@ public sealed class SqlServerFhirModel : IHostedService
             return _resourceTypeToId[resourceTypeName];
         }
 
         public bool TryGetResourceTypeId(string resourceTypeName, out short id)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _resourceTypeToId.TryGetValue(resourceTypeName, out id);
         }
 
         public string GetResourceTypeName(short resourceTypeId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _resourceTypeIdToTypeName[resourceTypeId];
         }
 
         public byte GetClaimTypeId(string claimTypeName)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _claimNameToId[claimTypeName];
         }
 
         public short GetSearchParamId(Uri searchParamUri)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _searchParamUriToId[searchParamUri];
         }
 
         public byte GetCompartmentTypeId(string compartmentType)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _compartmentTypeToId[compartmentType];
         }
 
         public bool TryGetSystemId(string system, out int systemId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _systemToId.TryGetValue(system, out systemId);
         }
 
         public int GetSystemId(string system)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
 
             VLatest.SystemTable systemTable = VLatest.System;
             return GetStringId(_systemToId, system, systemTable, systemTable.SystemId, systemTable.Value);
         }
 
         public int GetQuantityCodeId(string code)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
 
             VLatest.QuantityCodeTable quantityCodeTable = VLatest.QuantityCode;
             return GetStringId(_quantityCodeToId, code, quantityCodeTable, quantityCodeTable.QuantityCodeId, quantityCodeTable.Value);
         }
 
         public bool TryGetQuantityCodeId(string code, out int quantityCodeId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _quantityCodeToId.TryGetValue(code, out quantityCodeId);
         }
 
-        public async Task StartAsync(CancellationToken cancellationToken)
+        public Task EnsureInitialized()
         {
-            await _schemaInitializer.StartAsync(cancellationToken);
+            ThrowIfCurrentSchemaVersionIsNull();
 
-            if (_searchParameterDefinitionManager is IHostedService hostedService)
+            // If the fhir-server is just starting up, synchronize the fhir-server dictionaries with the SQL database
+            Initialize((int)_schemaInformation.Current, true);
+
+            return Task.CompletedTask;
+        }
+
+        public void Initialize(int version, bool runAllInitialization)
```

Comment:
>Initialize [](start = 20, length = 10)

Just curious, how it's protected from been called multiple times from different threads?
Are we just claim it's not thread-safe and know it would never be called concurrently?

Reply:
Yes, I believe that we are claiming that this is not thread-safe. I'm not sure if we have plans to look into this - @brendankowitz do you know if this is something on the roadmap?